docs/source/developer/release.rst

```diff
@@ -0,0 +1,97 @@
+
+
+Minor Version Release Candidates
+--------------------------------
+
+This is for first minor version release candidates. I.e. the first release of
+every series. It is special, because during this release, the release branch
+will have to be created. Release candidate indexing begins at 0.
+
+.. code-block::
+
+    ## Numba X.Y.Z
+
+    * [ ] merge to master:
+        - [ ] "remaining Pull-Requests from milestone"
+    * [ ] merge change log changes
+        - [ ] "PR with changelog entries
+    * [ ] Create X.Y release branch
+    * [ ] pin llvmlite to `>=0.A.0rc1,<0.A+1.0`
+    * [ ] annotated tag X.Y.Zrc0 on release branch
+    * [ ] build and upload conda packages on buildfarm (check "upload")
+    * [ ] build wheels (`$PYTHON_VERSIONS`) on the buildfarm
+    * [ ] upload wheels and sdist to PyPI (upload from `ci_artifacts`)
+    * [ ] verify packages uploaded to Anaconda Cloud and move to `numba/label/main`
+    * [ ] verify wheels for all platforms arrived on PyPi
+    * [ ] verify ReadTheDocs build
+    * [ ] clean up `ci_artifacts`
+    * [ ] review, merge and check execution of release notebook
+    * [ ] send RC announcement email / post announcement to discourse group
+    * [ ] post link to Twitter
+    * [ ] post link to python-announce-list@python.org
+
+    ### Post Release:
+
+    * [ ] tag X.Y+1.0dev0 to start new development cycle on `master`
+    * [ ] update llvmlite dependency spec to match next version via PR to `master`
+    * [ ] update release checklist template
+    * [ ] close milestone (and then close this release issue)
+
+Second Releases Candidates, Final Releases and Patch Releases
+-------------------------------------------------------------
+
+A patch release usually involves a series of cherry-picks, so the recipe is
+slightly different.
+
+.. code-block::
+
+    ## numba X.Y.Z
+
+    * [ ] cherry-pick items from the X.Y.Z milestone into a PR
+    * [ ] merge change log modifications and cherry-picks to X.Y release branch
+    * [ ] https://github.com/numba/numba/pull/XXXX
+    * [ ] annotated tag X.Y.Z on release branch (no `v` prefix)
+    * [ ] build and upload conda packages on buildfarm (check "upload")
+    * [ ] build wheels (`$PYTHON_VERSIONS`) on the buildfarm
+    * [ ] upload wheels and sdist to PyPI (upload from `ci_artifacts`)
+    * [ ] verify packages uploaded to Anaconda Cloud and move to `numba/label/main`
+    * [ ] verify wheels for all platforms arrived on PyPi
+    * [ ] verify ReadTheDocs build
+    * [ ] clean up `ci_artifacts`
+    * [ ] send RC/FINAL announcement email / post announcement to discourse group
+    * [ ] post link to Twitter
+    * [ ] post link to python-announce-list@python.org
+
+    ### Post release
+
+    * [ ] cherry-pick change-log modifications to main branch (`master`)
+    * [ ] update release checklist template
+    * [ ] ping Anaconda Distro team to trigger a build for `defaults` (FINAL ONLY)
+    * [ ] close milestone (and then close this release issue)
```

Comment:
Could these just be actual templates?

Reply:
In this directory: https://github.com/numba/numba/tree/master/.github/ISSUE_TEMPLATE

Reply:
Maybe. However, I am not sure we can git version control the templates then. I think keeping track of how they evolve might be useful. Maybe we can update the templates from the documentation directly?

Reply:
Maybe I can symlink the templates into Sphinx, let me try that.

Reply:
The templates are versioned by virtue of them being in the repo? You can see e.g. blame from numerous changes to the bug report template here:
https://github.com/numba/numba/blame/master/.github/ISSUE_TEMPLATE/Bug_report.md

or do you mean something else?

Reply:
> The templates are versioned by virtue of them being in the repo? You can see e.g. blame from numerous changes to the bug report template here:
> https://github.com/numba/numba/blame/master/.github/ISSUE_TEMPLATE/Bug_report.md
> 
> or do you mean something else?

I wasn't aware that this is how tempaltes worked. I though they were simply attached to the repo in an unversioned form. Thank you for clearing this up.

Reply:
No problem. What's also quite nice about maybe doing it this way is that you don't have to expose the template in the https://github.com/numba/numba/blob/master/.github/ISSUE_TEMPLATE/config.yml (as it's not useful to users) but I think it's still accessible directly as needed. E.g. by accessing:
```
https://github.com/numba/numba/issues/new?template=THE_RELEASE_TEMPLATE.md
```

Reply:
That is useful, I have included such URLs for both templates.